envconfig.go

```diff
@@ -240,30 +256,31 @@ func generateFieldHandler(i int, tag envTag, typeHandler FieldTypeHandler) func(
 		if tag.Name != "" {
 			var ev string
 			if ev, found = lookup(tag.Name); found {
-				val, err = typeHandler.Parsers[parser](ev)
+				val, err = typeHandler.Parsers[parser](expand(ev, lookup))
```

Comment:
This is a little weird to me.

If I set
```bash
V1='foo'
V2='${V1}' # note the single-quotes, not double-quotes
```
It'd be very surprising to me that a program looking at `V2` would evaluate it as `foo`.

Reply:
> I don't see the recursion. `ev` origins from `lookup` which is the original environment lookup. It's expanded once on line 259, and then passed on to the parser. The resulting `val` is never expanded.

I guess calling this recursion isn't quite right, as it doesn't keep calling itself.

But in my above example, `ev` will be set to `${V1}` (which should be the terminal value), and then `expand` would return `foo` (which is wrong).

I believe this line should just be `val, err = typeHandler.Parsers[parser](ev)`.

Reply:
Ah, so you want the expansion to be limited to the defaults. No expansion on values that we lookup?

Reply:
As a user, I'd find that surprising, because then we treat the defaults different from the actual values.

Reply:
Nah, you're right. Changing.